### PR TITLE
test: Adding missing tests or correcting existing tests

### PR DESCRIPTION
**Pull request by Pavan-Microsoft**
## Purpose
This pull request adds comprehensive unit tests for several backend modules, improves test coverage for authentication utilities, Azure Functions, and utility functions, and refines the Python test coverage configuration in CI. The key changes are grouped below.

### New and Improved Unit Tests

**Authentication Utilities:**
- Added `test_auth_utils.py` with thorough unit tests for `get_authenticated_user_details` and `get_tenantid`, covering production and development modes, header extraction, error handling, and edge cases.

**Admin Module:**
- Added `test_admin.py` with detailed tests for the `load_css` function, including file reading, error handling, and Streamlit integration, using extensive mocking to isolate dependencies.

**Azure Function:**
- Added `test_combine_pages_chunknos.py` with unit tests for the `combine_pages_and_chunknos` Azure Function, testing normal operation, error handling, edge cases, and response format compliance.

**App Utilities and Citation Handling:**
- Expanded and renamed the app test module to `test_create_app.py` (from `test_app.py`), adding new tests for `get_markdown_url` and `get_citations` utility functions. These tests cover URL formatting, SAS token handling, citation metadata extraction, and edge cases with various input formats. [[1]](diffhunk://#diff-2a885c4617d399237673e34aa7c5c323d5def99db250af5f44a6c30cd6145815L12-R12) [[2]](diffhunk://#diff-2a885c4617d399237673e34aa7c5c323d5def99db250af5f44a6c30cd6145815R1152-R1352)

### CI and Coverage Improvements

- Updated the Python test command in `.github/workflows/tests.yml` to use `--cov=code` (limiting coverage to the `code` directory), added `--cov-report=term-missing` for improved output, and adjusted the coverage threshold logic.

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -90,7 +90,7 @@ jobs:
 
           echo "MIN_COVERAGE=$MIN_COVERAGE" >> "$GITHUB_OUTPUT"
       - name: Run Python Tests
-        run: make python-test optional_args="--junitxml=coverage-junit.xml --cov=. --cov-report xml:coverage.xml --cov-fail-under ${{ steps.coverage-value.outputs.MIN_COVERAGE }} ./code/tests"
+        run: make python-test optional_args="--junitxml=coverage-junit.xml --cov=code --cov-report=term-missing --cov-report=xml:coverage.xml --cov-fail-under=${{ steps.coverage-value.outputs.MIN_COVERAGE }} ./code/tests"
```
[the remark this answers is not in the document]
Comment (Pavan-Microsoft) [reply .github/workflows/tests.yml:94]: correct configuration.

**Review by Roopan-Microsoft — APPROVED** [review]
